### DISCUSSION TITLE
Privatkopiera fungerar nu i Firefox för Android

### DISCUSSION THREAD

**Discussion by stefansundin** [stefansundin/privatkopiera#209] [General]
Jag har precis släppt v0.4.1 och nu kan man använda Privatkopiera i Firefox för Android. Så du kan privatkopiera i mobilen utan dator. Häftigt. 😎

Istället för FFmpeg så måste du använda en app som kan sköta nedladdningen. Jag rekommenderar appen Seal. Den finns att ladda ned från [GitHub](https://github.com/JunkFood02/Seal) eller [F-Droid](https://f-droid.org/packages/com.junkfood.seal/).

> [!WARNING]  
> Appen Seal finns **INTE** på Google Play Store. Det finns en fejk-app med samma namn och ikon men den är inte officiell. Använd på egen risk!!

Efter att du har lyckats installera Seal så kan du installera Privatkopiera i Firefox och använda det som vanligt. Istället för att klippa och klistra in i en kommandotolk så klistrar du in i Seal istället.

Nackdelar:
- Du kan inte ställa in filnamnet. Filen blir döpt till något kryptiskt som t.ex. `hls-ts-avc.mp4` eller bara `.m3u8`. Du får döpa om filen efter att den har laddats ned.
- Du kan inte välja kvalitén i Privatkopiera utan du får göra det i Seal istället.

Jag gjorde en skärminspelning så ni kan se hur enkelt det är:

https://github.com/stefansundin/privatkopiera/assets/1991151/79172a36-e1f2-4d31-b716-1926a78f10d1

Filen sparas i `Downloads/Seal/`.

Jag testade också med längre videoklipp och det fungerade bra.

Det här fungerar ju självklart inte i SVT Play-appen och liknande. Använd Dela-funktionen för att öppna samma video i Firefox!

Om ni har problem eller frågor så skriv gärna nedanför.

**Comment by delit** [comment]
Grymt jobbat! 

Finns det någon fördel med Privatkopieras kod/kommando som man ska kopiera från pluginet? För jag testade att kopiera webbläsarens URL direkt från SVT in i Seal, och det fungerade felfritt, samt fick korrekt filnamn.

**Reply by stefansundin** [reply]
Det var kanske dåligt exempel att jag använde SVT Play i min video. Seal använder yt-dlp som har inbyggt stöd för många sidor, så om du laddar ner från någon av dessa sidor så behövs inte Privatkopiera. Jag hittade den här listan: https://github.com/yt-dlp/yt-dlp/blob/master/supportedsites.md

Det finns andra sidor som inte stöds direkt av yt-dlp och då kan du använda Privatkopiera som jag visade. Det finns också sidor som kräver inloggning och Seal har ingen funktion att logga in på en sida, så då måste du också använda Privatkopiera.

**Reply by Phiplex** [reply]
Eftersom flera beskrivit fel med svenska undertexter, vill jag nämna något jag observerat då jag tittat på film från SVT Play mha en Apple TV. Dvs situationen har inget alls med Privatkopiera att göra. Efter en stunds tittande med svenska undertexter försvinner texten helt. Upptäckte det först efter ett tag, då jag ändå förstod det engelska ljudet, men textremsan var borta. Fick byta till undertexter på annat språk (t.ex norska) och sedan tillbaka till svenska varvid den återkom. Detta kunde sedan upprepa sig ett antal gånger under hela filmen. Vill endast nämna detta då det verkar vara ett allmänt fel med svenska undertexter (som alltså inte har med Privatkopiera alls att göra). Också osäker på om det har något med Apple TV enheten att göra eller om felet finns hos SVT Play.

**Comment by GitNinja2024** [comment]
Hej där,

Jag vill börja med att tacka dig för den senaste uppdateringen. Tyvärr har jag stött på problem med undertexterna på SVT. Jag har försökt att ladda ner dem på nytt och även försökt att återställa både tillägget och ffmpeg flera gånger, men problemet kvarstår. Har du några idéer om hur jag kan lösa detta med undertexter?

**Reply by stefansundin** [reply]
Se till att du använder FFmpeg från https://github.com/yt-dlp/FFmpeg-Builds/blob/master/README.md. Instruktionerna på hemsidan är uppdaterade sedan ett tag tillbaka och rekommenderar nu den. Om du fortfarande har problem så läs https://github.com/stefansundin/privatkopiera/issues/120.

Undertexter är orelaterat till Android så håll diskussionen i rätt tråd, tack!
